assets/src/edit-story/components/dropDown/index.js

```diff
@@ -0,0 +1,362 @@
+  position: absolute;
+  top: 100%;
+  left: 0;
+  z-index: 1000;
+  display: flex;
+  flex-direction: column;
+  flex-wrap: wrap;
+  float: left;
+  min-width: 160px;
+  width: 100%;
+  padding: 5px 0;
+  margin: 2px 0 0;
+  font-size: 14px;
+  text-align: left;
+  list-style: none;
+  background-color: ${({ theme }) => theme.colors.fg.v1};
+  background-clip: padding-box;
+  border-radius: 4px;
+  box-shadow: 0 6px 12px ${({ theme }) => rgba(theme.colors.bg.v0, 0.175)};
+`;
+
+const DropDownItem = styled.li.attrs({ tabIndex: '0', role: 'option' })`
+  letter-spacing: ${({ theme }) => theme.fonts.label.letterSpacing};
+  padding: 16px;
+  margin: 0;
+  font-family: ${({ theme }) => theme.fonts.label.family};
+  font-size: ${({ theme }) => theme.fonts.label.size};
+  line-height: ${({ theme }) => theme.fonts.label.lineHeight};
+  font-weight: ${({ theme }) => theme.fonts.label.weight};
+
+  &:hover {
+    background-color: ${({ theme }) => theme.colors.bg.v11};
+  }
+
+  &:focus {
+    background-color: ${({ theme }) => theme.colors.bg.v11};
+    outline: none;
+  }
+`;
+
+const possibleNormalKeysForSearch = [
+  'a',
+  'b',
+  'c',
+  'd',
+  'e',
+  'f',
+  'g',
+  'h',
+  'i',
+  'j',
+  'k',
+  'l',
+  'm',
+  'n',
+  'o',
+  'p',
+  'q',
+  'r',
+  's',
+  't',
+  'u',
+  'v',
+  'w',
+  'x',
+  'y',
+  'z',
+  '1',
+  '2',
+  '3',
+  '4',
+  '5',
+  '6',
+  '7',
+  '8',
+  '9',
+  '0',
+];
+
+function DropDown({ options, value, onChange, disabled }) {
+  DropDown.wrapperRef = useRef(null);
+  DropDown.selectRef = useRef();
+  DropDown.arrayOfOptionsRefs = [];
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
+  const [focusedValue, setFocusedValue] = useState(null);
+  const focusedIndex = useMemo(
+    () => options.findIndex((item) => item.value === focusedValue),
+    [focusedValue, options]
+  );
+  const activeItem = useMemo(
+    () => options.find((item) => item.value === value),
+    [value, options]
+  );
+  const toggleOptions = useCallback(() => {
+    setIsOpen(false);
+  }, []);
+
+  const handleUpDown = useCallback(
+    ({ key }) => {
+      if (!isOpen) {
+        setIsOpen(true);
+        setFocusedValue(value);
+      } else if (key === 'ArrowUp' && focusedIndex !== 0) {
+        handleMoveFocus(-1);
+      } else if (key === 'ArrowDown' && focusedIndex < options.length - 1) {
+        handleMoveFocus(1);
+      }
+    },
+    [isOpen, value, focusedIndex, options, handleMoveFocus]
+  );
+
+  const handleMoveFocus = useCallback(
+    (offset) => {
+      const findIndex = options.findIndex(
+        (item) => item.value === focusedValue
+      );
+      if (findIndex + offset >= 0 && findIndex + offset < options.length) {
+        setFocusedValue(options[findIndex + offset].value);
+      }
+    },
+    [focusedValue, options]
+  );
+
+  const handleKeyDown = useCallback(
+    ({ keyCode }) => {
+      const searchTerm = searchValue + String.fromCharCode(keyCode);
```

Comment:
Curious: Why is it necessary to add the `searchValue` and `String.fromCharCode(keyCode)` together here?

Reply:
Ah, okay, I see now, it's not just one letter search anymore.